pennylane/spin/spin_hamiltonian.py

```diff
@@ -309,3 +309,308 @@ def fermi_hubbard(
+    :math:`n_{i \uparrow}`, :math:`n_{i \downarrow}` are number operators for spin-up and
+    spin-down fermions at site ``i``, and :math:`n_{i}`, :math:`n_{j}` are number operators at sites ``i`` and ``j``.
+    This function assumes there are two fermions with opposite spins on each lattice site.
+
+    Args:
+        lattice (str): Shape of the lattice. Input values can be ``'chain'``, ``'square'``,
+            ``'rectangle'``, ``'honeycomb'``, ``'triangle'``, or ``'kagome'``.
+        n_cells (list[int]): Number of cells in each direction of the grid.
+        hopping (float | list[float] | list[math.array(float)]): Hopping strength between
+            neighbouring sites. It can be a number, a list of length equal to ``neighbour_order`` or
+            a square matrix of size ``(num_spins, num_spins)``, where ``num_spins`` is the total
+            number of spins. Default value is 1.0.
+        coulomb (float | list[float]): Coulomb interaction between spins. It can be a constant or a
+            list of length equal to number of spins.
+        intersite_coupling (float | list[float] | list[math.array(float)]): Interaction strength between spins on
+            neighbouring sites. It can be a number, a list with length equal to ``neighbour_order`` or
+            a square matrix of size ``(num_spins, num_spins)``, where ``num_spins`` is the total
+            number of spins. Default value is 1.0.
+        boundary_condition (bool | list[bool]): Defines boundary conditions for different lattice
+            axes. Default is ``False`` indicating open boundary condition.
+        neighbour_order (int): Specifies the interaction level for neighbors within the lattice.
+            Default is 1, indicating nearest neighbours.
+        mapping (str): Specifies the fermion-to-qubit mapping. Input values can be
+            ``'jordan_wigner'``, ``'parity'`` or ``'bravyi_kitaev'``.
+
+    Returns:
+       ~ops.op_math.Sum: Hamiltonian for the Emery model.
+
+    **Example**
+
+    >>> n_cells = [2]
+    >>> h = [0.5]
+    >>> u = 1.0
+    >>> v = 0.2
+    >>> spin_ham = qml.spin.emery("chain", n_cells, hopping=h, coulomb=u, intersite_coupling=v)
+    >>> spin_ham
+    (
+    -0.25 * (Y(0) @ Z(1) @ Y(2))
+    + -0.25 * (X(0) @ Z(1) @ X(2))
+    + 0.7000000000000002 * I(0)
+    + -0.25 * (Y(1) @ Z(2) @ Y(3))
+    + -0.25 * (X(1) @ Z(2) @ X(3))
+    + -0.35 * Z(1)
+    + -0.35 * Z(0)
+    + 0.25 * (Z(0) @ Z(1))
+    + -0.35 * Z(3)
+    + -0.35 * Z(2)
+    + 0.25 * (Z(2) @ Z(3))
+    + 0.05 * (Z(0) @ Z(2))
+    + 0.05 * (Z(0) @ Z(3))
+    + 0.05 * (Z(1) @ Z(2))
+    + 0.05 * (Z(1) @ Z(3))
+    )
+
+    """
+
+    lattice = _generate_lattice(lattice, n_cells, boundary_condition, neighbour_order)
+
+    if isinstance(hopping, (int, float, complex)):
+        hopping = [hopping]
+
+    hopping = math.asarray(hopping)
+
+    if isinstance(intersite_coupling, (int, float, complex)):
+        intersite_coupling = [intersite_coupling]
```

Comment:
Personally i would flip this to just having one if statement at the top
e.g.
```suggestion
    if not isinstance(hopping, (int, float, complex)) or not isinstance(intersite_coupling, (int, float, complex)):
         raise TypeError(f"wrong type buddy")
         
    hopping = [hopping]
    hopping = math.asarray(hopping
    
    intersite_coupling = [intersite_coupling]
    intersite_coupling = math.asarray(intersite_coupling]
```

youd have to add another test case though 😅